src/actions/Entities/NameNoteActions.js

```diff
@@ -41,26 +42,41 @@ const saveNew = nameNote => ({
   payload: { nameNote },
 });
 
-const saveNewSurvey = surveyData => dispatch => {
+const saveNewSurvey = surveyData => (dispatch, getState) => {
   const nameNote = createDefaultNameNote();
+  const nameId = selectEditingNameId(getState());
   const patientEvents = UIDatabase.objects('PCDEvents');
 
   if (patientEvents.length === 0) return;
   const [patientEvent] = patientEvents;
 
   nameNote.data = surveyData;
   nameNote.patientEvent = patientEvent.id;
+  nameNote.name = nameId;
 
   dispatch({
     type: NAME_NOTE_ACTIONS.SAVE_NEW,
     payload: { nameNote },
   });
 };
 
-const saveEditing = nameNote => ({
-  type: NAME_NOTE_ACTIONS.SAVE_EDITING,
-  payload: { nameNote },
-});
+const saveEditing = () => (dispatch, getState) => {
+  const currentNameNote = selectEditingNameNote(getState());
+  const names = UIDatabase.objects('Patient').filtered('id = $0', currentNameNote.name);
+  const patientEvents = UIDatabase.objects('PCDEvents').filtered(
+    'id = $0',
+    currentNameNote.patientEvent
+  );
+
+  if (names.length && patientEvents.length) {
+    const name = names[0];
+    const patientEvent = patientEvents[0];
```

Comment:
there's a better way, surely?

Reply:
Just using a `.get()` I would have thought.. tried that first and it wasn't working but I think it might have been an error caused by something else